### ISSUE TITLE
Auth and Guest middleware redirects when fails

### ISSUE THREAD

**Issue by hiendv** [laravel/framework#11610]
Am I wrong to say that these redirects should be related to authentication guards somehow?
`Auth::guard($guard)->check()` obviously depends on `$guard`, so why just put `redirect('/')` there  instead of something related to that `$guard`, e.g `redirect(Auth::guard($guard)->redirects['check'])` which make the redirection more meaningful?
For example I have two drivers `frontend` and `backend`. That would be great to have `redirects` configuration on each of them.
```php
'guards' => [
        'frontend' => [
            'driver' => 'session',
            'provider' => 'users',
            'redirects' => [
                'guest' => '/login',
                'check' => '/'
            ]
        ],

        'backend' => [
            'driver' => 'session',
            'provider' => 'admins',
            'redirects' => [
                'guest' => route('backend.getLogin'),
                'check' => '/backend'
            ]
        ],
]
```
In conclusion, I'm talking about the default behaviour of authentication middlewares when failing that should be attached to guards instead of `'/'` or `'/login'`. Hope to hear your thought soon !

**Comment by vinicius73** [comment]
I think this would not be his job, but the middleware. how is today.

**Comment by hiendv** [comment]
I think there should be some kinds of default behaviours when failing.

**Comment by vinicius73** [comment]
anyway, it is a setting you can make q no impact at all.
You need only retrieve these values in your middleware.

**Comment by GrahamCampbell** [comment]
I think it's best to just write your own middleware to do exactly what you want.

**Comment by hiendv** [comment]
I got it but can't help questioning myself about the role of guards in authentication. Authentication guard is a great feature but it isn't implemented much in the framework. Anyway, thank you !
